install_config/advanced_ldap_configuration/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,425 @@
+authentication request is issued for the *openshift* stack.
+
+. Configure the Apache *_httpd.conf_*. The steps in this section focus on
+setting up the challenge authentication, which is useful for logging in with `oc
+login` and similar automated tools.
++
+[NOTE]
+====
+link:../advanced_ldap_configuration/configuring_form_based_authentication.html[Configuring
+Form-Based Authentication] explains how you can get a template login screen.
+====
+
+. Create the new file *_openshift-proxy.conf_* in *_/etc/httpd/conf.d_*
+(substituting the correct host names where indicated):
++
+====
+----
+LoadModule request_module modules/mod_request.so
+LoadModule lookup_identity_module modules/mod_lookup_identity.so
+# Nothing needs to be served over HTTP.  This virtual host simply redirects to
+# HTTPS.
+<VirtualHost *:80>
+  DocumentRoot /var/www/html
+  RewriteEngine              On
+  RewriteRule     ^(.*)$     https://%{HTTP_HOST}$1 [R,L]
+</VirtualHost>
+
+<VirtualHost *:443>
+  # This needs to match the certificates you generated.  See the CN and X509v3
+  # Subject Alternative Name in the output of:
+  # openssl x509 -text -in /etc/pki/tls/certs/proxy.example.com.crt
+  ServerName proxy.example.com
+
+  DocumentRoot /var/www/html
+  SSLEngine on
+  SSLCertificateFile /etc/pki/tls/certs/proxy.example.com.crt
+  SSLCertificateKeyFile /etc/pki/tls/private/proxy.example.com.key
+  SSLCACertificateFile /etc/pki/CA/certs/ca.crt
+
+  # Send logs to a specific location to make them easier to find
+  ErrorLog logs/proxy_error_log
+  TransferLog logs/proxy_access_log
+  LogLevel warn
+  SSLProxyEngine on
+  SSLProxyCACertificateFile /etc/pki/CA/certs/ca.crt
+  # It's critical to enforce client certificates on the Master.  Otherwise
+  # requests could spoof the X-Remote-User header by accessing the Master's
+  # /oauth/authorize endpoint directly.
+  SSLProxyMachineCertificateFile /etc/pki/tls/certs/authproxy.pem
+
+  # Send all requests to the console
+  RewriteEngine              On
+  RewriteRule     ^/console(.*)$     https://%{HTTP_HOST}:8443/console$1 [R,L]
+
+  # In order to using the challenging-proxy an X-Csrf-Token must be present.
+  RewriteCond %{REQUEST_URI} ^/challenging-proxy
+  RewriteCond %{HTTP:X-Csrf-Token} ^$ [NC]
+  RewriteRule ^.* - [F,L]
+
+  <Location /challenging-proxy/oauth/authorize>
+    # Insert your backend server name/ip here.
+    ProxyPass https://openshift.example.com:8443/oauth/authorize
+    AuthType Basic
+    AuthBasicProvider PAM
+    AuthPAMService openshift
+    Require valid-user
+  </Location>
+
+  <ProxyMatch /oauth/authorize>
+    AuthName openshift
+    RequestHeader set X-Remote-User %{REMOTE_USER}s
+  </ProxyMatch>
+</VirtualHost>
+
+RequestHeader unset X-Remote-User
+----
+====
++
+[NOTE]
+====
+link:../advanced_ldap_configuration/configuring_form_based_authentication.html[Configuring
+Form-Based Authentication] explains how to add the *login-proxy* block to
+support form authentication.
+====
```

Comment:
Not sure how to address this change just yet. The current wording fits our style guidelines and a "it's coming later" statement does not make sense in the current structure because the structure does not necessarily indicate that the separate topics should be a procedure followed in a particular order.  :confused: